web/service-generator/src/index.ts

```diff
@@ -0,0 +1,39 @@
+import { createEcmaScriptPlugin } from '@bufbuild/protoplugin';
+import { GeneratedFile, Schema, findCustomMessageOption } from '@bufbuild/protoplugin/ecmascript';
+import { DescMethod } from '@bufbuild/protobuf';
+import { version } from '../package.json';
+import { HttpRule } from './http_pb';
+
+export default createEcmaScriptPlugin({
+  name:    'service-generator',
+  version: `v${ String(version) }`,
+  generateTs,
+});
+
+function generateTs(schema: Schema) {
+  schema.files.forEach((file) => {
+    const f = schema.generateFile(`${ file.name }_svc.ts`);
+
+    f.print('import axios from "axios";\n');
+
+    file.services.forEach((service) => {
+      service.methods.forEach(method => printMethod(f, method));
+    });
+  });
+}
+
+function printMethod(f: GeneratedFile, method: DescMethod) {
+  const m = findCustomMessageOption(method, 72295728, HttpRule);
+  const input = f.import(method.input);
+  const output = f.import(method.output);
+
+  f.print(`export async function ${ method.name }(input:`, input, `): Promise<`, output, `> {
```

Comment:
The generated code preamble comments should be added here to match protoc-gen-es

Reply:
Done